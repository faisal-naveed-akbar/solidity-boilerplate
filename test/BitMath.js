
const assert = require("assert");
const chai = require('chai')
require('mocha');
const { jestSnapshotPlugin } = require("mocha-chai-jest-snapshot");
chai.use(jestSnapshotPlugin()); 
const { expect } = require("chai");
const BitMathTest = artifacts.require("BitMathTest");

contract("BitMathTest", accounts => {
    
    describe("MostSignificantBit", function () {
        
        it("should return 0 as msb on passing 1", async () => {
            const instance = await BitMathTest.deployed();
            await assert.doesNotReject(() => instance.mostSignificantBit(1));
        });

        it("should return 1 as msb on passing 2", async () => {
            const instance = await BitMathTest.deployed();
            await assert.doesNotReject(() => instance.mostSignificantBit(2));
        });

        it("should return 2 as msb on passing 4", async () => {
            const instance = await BitMathTest.deployed();
            await assert.doesNotReject(() => instance.mostSignificantBit(4));
        });

        it("should return 4 as msb on passing 16", async () => {
            const instance = await BitMathTest.deployed();
            await assert.doesNotReject(() => instance.mostSignificantBit(16));
        });
        
        it("should return 8 as msb on passing 256", async () => {
            const instance = await BitMathTest.deployed();
            await assert.doesNotReject(() => instance.mostSignificantBit(256));
        });

        it("should return 16 as msb on passing 65536", async () => {
            const instance = await BitMathTest.deployed();
            await assert.doesNotReject(() => instance.mostSignificantBit(65536));
        });

        it("should return 16 as msb on passing 65536", async () => {
            const instance = await BitMathTest.deployed();
            await assert.doesNotReject(() => instance.mostSignificantBit(4294967296));
        });

        it("should use same gas price for every msb calculation", async () => {
            const instance = await BitMathTest.deployed();
            const gas =  await instance.getGasCostOfMostSignificantBit(2);
            //expect({ gas }).toMatchSnapshot();
        });
        
    
    });

});