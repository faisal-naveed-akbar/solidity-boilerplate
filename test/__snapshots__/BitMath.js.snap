// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contract: BitMathTest MostSignificantBit should use same gas price for every msb calculation 1`] = `
Object {
  "gas": Object {
    "logs": Array [],
    "receipt": Object {
      "blockHash": "0x556abd469ba28290709f51b5f350a9804c9d4125102704b9fa182269a0deade4",
      "blockNumber": 15,
      "contractAddress": null,
      "cumulativeGasUsed": 23084,
      "from": "0x545234ba61afec3eac600ff1f62e4c2983369ee6",
      "gasUsed": 23084,
      "logs": Array [],
      "logsBloom": "0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000",
      "rawLogs": Array [],
      "status": true,
      "to": "0xd6e19c0817c20358abf019e5db7ed38bcf397bb2",
      "transactionHash": "0x5c8992e34bca564a99595f75da9cad3cc781ded16f4e51fe1fc1ba75c8028597",
      "transactionIndex": 0,
    },
    "tx": "0x5c8992e34bca564a99595f75da9cad3cc781ded16f4e51fe1fc1ba75c8028597",
  },
}
`;
